packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -55,9 +83,47 @@ const ShowHide = (props) => {
     preview,
     intl,
     isOpen,
+    focusRef,
     ...customProps
   } = props;
 
+  const theme = useContext(ThemeContext);
+  const contentRef = useRef(null);
+  const [containerIsActive, setContainerIsActive] = useState(false);
+
+  // Keep track of modified attributes to remove on blur
+  const isModified = useRef({
+    tabIndex: false,
+    role: false,
+  });
+
+  /**
+   * Upon showing hidden content, set activeElement to focusRef element if provided, othervise set it to content container.
+   */
+  useEffect(() => {
+    if (isOpen) {
+      if (focusRef?.current) {
+        if (checkIfAddTabindex(focusRef?.current)) {
+          // not focusable element with no tabindex set, safe to temporary add tabindex that will later be removed on blur
+          focusRef?.current.setAttribute('tabIndex', '-1');
+          isModified.current.tabIndex = true;
+          if (!focusRef?.current.hasAttribute('role')) {
+            focusRef?.current.setAttribute('role', 'group');
+            isModified.current.role = true;
+          }
+        }
+        focusRef?.current.focus();
+      } else if (contentRef?.current) {
+        setContainerIsActive(true);
+      }
+    }
+  }, [isOpen, focusRef, contentRef]);
```

Comment:
Not a fan of having the whole function body be an `if` block. Can we change this to something like:

```js
if(!isOpen)
  return;

// rest of the function

```

This will reduce at least 1 nested `if`.

Reply:
Fixed in this commit: https://github.com/cerner/terra-core/pull/3838/commits/4d398feeb8d9d23ccb3669c45292949bfaaa0ff3